c_glib/arrow-glib/record-batch.cpp

```diff
@@ -152,7 +171,12 @@ garrow_record_batch_new(GArrowSchema *schema,
   auto arrow_record_batch =
     arrow::RecordBatch::Make(garrow_schema_get_raw(schema),
                              n_rows, arrow_columns);
-  return garrow_record_batch_new_raw(&arrow_record_batch);
+  auto status = arrow_record_batch->Validate();
```

Comment:
This also checks the schema types match the field types. If that becomes too expensive we might add a faster validation function

Reply:
OK. I'll create a JIRA ticket when it's too expensive.